subsys/net/lib/config/configuration.yaml

```diff
@@ -0,0 +1,117 @@
+# This is an example for defining network configuration data that
+# can be applied automatically when the device boots.
+#
+net:
+ network-interfaces:
+   # Example of one interface selected by its name
+   - &main-interface
+     name: eth0
+     set-name: my-eth0
+     set-default: True
+     IPv6:
+       status: enabled
+       addresses:
+       - 2001:db8:110::1
+       multicast-addresses:
+       - ff05::114
+       - ff15::115
+       prefixes:
+         - address: "2001:db8::"
+           len: 64
+           lifetime: 1024
+       hop-limit: 64
+       multicast-hop-limit: 1
+       DHCPv6:
+         status: enabled
+         do-request-address: true
+         do-request-prefix: false
+     IPv4:
+       status: enabled
+       addresses:
+       - 192.0.2.10
+       multicast-addresses:
+       - 234.0.0.10
+       gateway: 192.0.2.1
+       netmask-len: 24
+       time-to-live: 64
+       multicast-time-to-live: 1
+       DHCPv4: enabled
+       IPv4-autoconf: disabled
```

Comment:
Please note that this is just an example file, no one in their sane mind would use static IP address in the production device. But we need to set these for testing purposes, that is why you see various static IP addresses in our prj.conf files in the `tests/net` and `samples/net`. This PR is nothing more than just allowing us to configure the device with more than one network interface, all of this is already in use for one network interface (using Kconfig variables).

Reply:
Also note that this PR is not about provisioning the device by the vendor. It allows that kind of use, but user needs to be very careful about that as you noticed.

Reply:
A test can just set it up manually, I see no need for this. What _would_ be useful is if there was a system on the device itself that it could set/use settings for interfaces, e.g. dhcpcd config from linux but using settings system, that would be really useful

Reply:
Shameless plug to my comment [here](https://github.com/zephyrproject-rtos/zephyr/pull/68127#issuecomment-1911734081), where a concise binary representation would help here.

Scenario **Runtime update**: encode and write it to a settings backend, or littlefs and read it back after cold boot/wakeup. 

Even supports DFU as the data can be decoded in a backwards compatible way.

Reply:
That would be brilliant because basically anyone doing this now in zephyr has to do it from scratch (as I have done previously) so to have such a feature would be useful for anyone using networking in zephyr - applications, tests and samples

Reply:
> Please note that this is just an example file, no one in their sane mind would use static IP address in the production device.

Hehehe... I know some people who would do that.. oh sane, nm 😅